policy-controller/grpc/src/outbound.rs

```diff
@@ -202,80 +204,84 @@ fn to_service(outbound: OutboundPolicy) -> outbound::OutboundPolicy {
+
+                outbound::proxy_protocol::Kind::Detect(outbound::proxy_protocol::Detect {
+                    timeout: Some(
+                        time::Duration::from_secs(10)
+                            .try_into()
+                            .expect("failed to convert detect timeout to protobuf"),
+                    ),
+                    opaque: Some(outbound::proxy_protocol::Opaque {
+                        routes: vec![default_outbound_opaq_route(backend)],
+                    }),
+                    http1: Some(outbound::proxy_protocol::Http1 {
+                        routes: routes.clone(),
+                        failure_accrual: accrual.clone(),
+                    }),
+                    http2: Some(outbound::proxy_protocol::Http2 {
+                        routes,
+                        failure_accrual: accrual,
+                    }),
+                })
+            }
+            Some(OutboundRouteCollection::Http(routes)) => {
+                let mut routes = routes
+                    .into_iter()
+                    .sorted_by(timestamp_then_name)
+                    .map(|(gknn, route)| convert_outbound_http_route(gknn, route, backend.clone()))
+                    .collect::<Vec<_>>();
 
-        if http_routes.is_empty() {
-            http_routes = vec![default_outbound_http_route(backend.clone())];
-        }
+                if routes.is_empty() {
+                    routes = vec![default_outbound_http_route(backend.clone())];
+                }
 
-        let accrual =
-            outbound
-                .accrual
-                .map(|accrual| linkerd2_proxy_api::outbound::FailureAccrual {
-                    kind: Some(match accrual {
-                        linkerd_policy_controller_core::outbound::FailureAccrual::Consecutive {
-                            max_failures,
-                            backoff,
-                        } => outbound::failure_accrual::Kind::ConsecutiveFailures(
-                            outbound::failure_accrual::ConsecutiveFailures {
-                                max_failures,
-                                backoff: Some(outbound::ExponentialBackoff {
-                                    min_backoff: convert_duration(
-                                        "min_backoff",
-                                        backoff.min_penalty,
-                                    ),
-                                    max_backoff: convert_duration(
-                                        "max_backoff",
-                                        backoff.max_penalty,
-                                    ),
-                                    jitter_ratio: backoff.jitter,
-                                }),
-                            },
-                        ),
+                // TODO(the-wondersmith): determine what (if any) contracts
+                //  would need to be updated if we return `Kind::Http` here
```

Comment:
That's what I thought as well, but also had a fuzzy memory of @olix0r mentioning that even though we've _historically_ returned either `Opaque` or `Detect` that we should have eyes toward returning more exact (i.e. non-`Detect`) responses if we're able to. 

It might be worth a conversation elsewhere about maybe unifying the `Http_` variants of [`linkerd2_proxy_api::outbound::proxy_protocol::Kind`](https://github.com/linkerd/linkerd2-proxy-api/blob/7c82fb177895466cde5a73e89de0bf27fab9f740/src/gen/io.linkerd.proxy.outbound.rs#L97), given that there's no real way (so far as I know) to differentiate from the policy controller's side of things? 

For the purposes of _this_ review/conversation, I'll remove the todo and mark it resolved seeing as it looks like we're in agreement.